wdl-engine/src/backend/crankshaft.rs

```diff
@@ -85,50 +104,83 @@ impl TaskManagerRequest for CrankshaftTaskRequest {
         self.memory
     }
 
-    async fn run(self, spawned: oneshot::Sender<()>) -> Result<i32> {
+    async fn run(self, spawned: oneshot::Sender<()>) -> Result<TaskExecutionResult> {
         // Create the working directory
-        let work_dir = self.inner.root.work_dir();
-        fs::create_dir_all(work_dir).with_context(|| {
+        // TODO: this should only be done for local task execution
+        let work_dir = self.inner.root.attempt_dir().join(WORK_DIR_NAME);
+        fs::create_dir_all(&work_dir).with_context(|| {
             format!(
                 "failed to create directory `{path}`",
                 path = work_dir.display()
             )
         })?;
 
         // Write the evaluated command to disk
+        // This is done even for remote execution so that a copy exists locally
         let command_path = self.inner.root.command();
-        fs::write(command_path, &self.inner.command).with_context(|| {
+        fs::write(command_path, self.inner.command()).with_context(|| {
             format!(
                 "failed to write command contents to `{path}`",
                 path = command_path.display()
             )
         })?;
 
-        let inputs = self
-            .inner
-            .mounts
-            .iter()
-            .filter(|m| m.host.exists())
-            .map(|m| {
-                Ok(Arc::new(
+        let mut inputs = Vec::with_capacity(self.inner.inputs().len() + 2);
```

Comment:
Can we document why `+ 2`?

Reply:
Now looking forward, I see the two entries. I'd still recommend a comment here, but that's a nit.